### PR TITLE
feat: add in pytorch base image

### PR DESCRIPTION
**Pull request by ahodges22**
Adding in a Python image that includes PyTorch and CUDA/cuDNN (PyTorch ships their own binaries for these). This will primarily be used for the Image Generation service, but may also be used for other services like Avatar API.

### CODE REVIEW

**Comment by ahodges22** [comment]
cc @tecnobrat @staturecrane

File: 3.12-pytorch/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM articulate/python:3.12
```
Comment (mloberg) [inline 3.12-pytorch/Dockerfile:1]: All images build at the same time, so we won't get the latest updates if we build off this

Comment (ahodges22) [reply 3.12-pytorch/Dockerfile]: Just updated it to not reference the existing Python image :+1:

**Review by mloberg — APPROVED** [review]
@ahodges22 I changed how we set platforms for the pytorch image. I also moved to the 3.12 directory to keep things a little more organized if we need to add more variants

**Pull request merged**